File: LeetCode Problems/1801 to 1900/1833. Maximum Ice Cream Bars/1833. Maximum Ice Cream Bars.cpp
class Solution {
public:
    int maxIceCream(vector<int>& costs, int coins) {
        sort(costs.begin(), costs.end());
        int sum = 0, cnt = 0;
        for(int i = 0; i < costs.size(); i++) {
            if(sum + costs[i] <= coins) {
                sum += costs[i];
                cnt++;
            }
        }
        return cnt;
    }
};

File: LeetCode Problems/1801 to 1900/1833. Maximum Ice Cream Bars/1833. Maximum Ice Cream Bars.java
class Solution {
    public int maxIceCream(int[] costs, int coins) {
        Arrays.sort(costs);
        int cost = 0, cnt = 0;
        for (int i = 0; i < costs.length; i++) {
            if (costs[i] + cost <= coins) {
                cost += costs[i];
                cnt++;
            }
        }
        return cnt;
    }
}
